student/src/main/java/tds/student/sql/repository/remote/impl/RemoteExamRepository.java

```diff
@@ -317,4 +317,31 @@ public void waitForSegmentApproval(final UUID examId, final SegmentApprovalReque
       throw new ReturnStatusException(rce);
     }
   }
+
+  @Override
+  public List<ExamAssessmentMetadata> findExamAssessmentInfo(final long studentId, final UUID sessionId, final String grade) throws ReturnStatusException {
+    HttpHeaders headers = new HttpHeaders();
+    headers.set("Accept", MediaType.APPLICATION_JSON_VALUE);
+    headers.setContentType(MediaType.APPLICATION_JSON);
+    HttpEntity<?> requestHttpEntity = new HttpEntity<>(headers);
+    ResponseEntity<Response<List<ExamAssessmentMetadata>>> responseEntity;
+
+    UriComponentsBuilder builder = UriComponentsBuilder.fromHttpUrl(String.format("%s/metadata", examUrl))
+        .queryParam("studentId", studentId)
+        .queryParam("sessionId", sessionId)
+        .queryParam("grade", grade);
+
+    try {
+      responseEntity = restTemplate.exchange(
+          builder.build().encode().toUri(),
+          HttpMethod.GET,
+          requestHttpEntity,
+          new ParameterizedTypeReference<Response<List<ExamAssessmentMetadata>>>() {
+          });
+    } catch (RestClientException rce) {
+      throw new ReturnStatusException(rce);
```

Comment:
We haven't been doing this but after dealing with all the issues in item selection I agree it is a good idea